Code/Framework/AzToolsFramework/Tests/Input/QtEventToAzInputMapperTests.cpp

```diff
@@ -512,4 +516,32 @@ namespace UnitTest
             return info.param.m_az.GetName();
         }
     );
+
+    TEST_F(QtEventToAzInputMapperFixture, MouseWrapMouseViewportQtEventToAzInputMapperFixture) 
+    {
+        AzFramework::InputChannelNotificationBus::Handler::BusConnect();
+        m_captureAzEvents = false;
+
+        
+        const auto startPos = QPoint(WidgetSize.width()  - 2, WidgetSize.height() / 2);
+        MouseMove(m_rootWidget.get(), startPos, QPoint(0,0));
+        m_inputChannelMapper->SetCursorMode(AzToolsFramework::QtEventToAzInputMapper::CursorInputMode::CursorModeWrappedX);
+
+        const auto deltaPos = QPoint(200.0f, 0);
+        const auto expectedPosition = m_rootWidget->mapToGlobal(QPoint(200, WidgetSize.height() / 2));
+        const int iterations = 50;
+  
+        for(float i = 0; i < iterations; i++) {
+            MouseMove(m_rootWidget.get(), m_rootWidget->mapFromGlobal(QCursor::pos()), (deltaPos / iterations));
+        }
```

Comment:
suggestion: In this example I don't think we need iterations, we just need a single mouse move to check moving past the edge of the screen wraps (for all the extremes ideally)